### ISSUE TITLE
NVDA does not announce certain gesture commands when using SAPI 5 voice with WASAPI audio output enabled

### ISSUE THREAD

**Issue by HBM301** [nvaccess/nvda#18797]
### Steps to reproduce:
Try switching to any SAPI 5 voice and test the following gestures:
- Toggle Screen Curtain: NVDA+Control+Escape:
- NVDA+Alt+R (when Remote Access is disabled)

and several other gestures that I haven't tested yet.
### Actual behavior:

NVDA does not announce anything when pressing the gestures mentioned above, even though the gestures themselves still function normally.
### Expected behavior:

NVDA should announce those gestures,  the issue is only resolved when the checkbox Use modern audio output system (WASAPI)" is unchecked.
#### NVDA installed/portable/running from source:
Tryed with both installed and portable
#### NVDA version:
2025.3 beta 2
#### Windows version:
Windows 11 24H2 (AMD64) build 26100.4946
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, This issue does not occur in NVDA 2025.2, and 2025.3 beta1
#### If NVDA add-ons are disabled, is your problem still occurring?
Don't install any addons
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tested

**Comment by seanbudd** [comment]
@gexgd0419 - appears to be a regression from #18675

**Comment by HBM2001** [comment]
Currently, I'm using a SAPI5-based voice (Sao Mai voice) With silence trimming enabled, the response has been significantly faster. However, since version 2025.3 beta, I’ve noticed a slight decrease in responsiveness and flexibility. This issue  did not occur in version 2025.2, I think the problem might be with the SAPI5 bug fix PRs for 2025.3

**Comment by gexgd0419** [comment]
As an attempt to fix #17967, I added a little bit of delay, in order to mitigate the audio gap problem in some voices.

When rate boost is enabled, the extra processing time may cause the voice to fail sending audio data in time. So I force the voice to prepare at least 50 ms of audio before playing, which increases the delay a little bit (depends on how much time the voice takes to prepare 50 ms of audio). Although I think that most voices should be fine without the delay when rate boost is off.

What should I do? @seanbudd

- Only add the delay when rate boost is on? This way it's still as responsive as before, but users may notice an increase in delay when turning on rate boost.
- Add an extra option for users to opt-in when they find audio gaps in their voices?
- Try to detect when audio gaps happen and enable this feature dynamically after that?

**Comment by gexgd0419** [comment]
@HBM301 Could you try [this snapshot build](https://github.com/nvaccess/nvda/actions/runs/17214090630/artifacts/3845373708)?

**Comment by HBM2001** [comment]
@gexgd0419
HI, I’ve tried your snapshot version, and it seems there’s no noticeable difference compared to the 2025.3 beta. In my opinion, the idea of adding an option for users to manually enable the feature when they experience audio gaps is a good one
I truly appreciate your silence  trimming feature, as it has significantly improved the response speed of the voice I'm using, Thanks

**Comment by gexgd0419** [comment]
@HBM301 Is the "gesture not announced" problem fixed in that version?

**Comment by HBM2001** [comment]
@gexgd0419 The gestures are now reporting normally again.

**Comment by seanbudd** [comment]
@gexgd0419 @HBM301 - can either of you please start a separate discussion/issue for the rate boost / delay issue which is separate to this issue for gestures not being announced
